middleware/redirect.go

```diff
@@ -0,0 +1,48 @@
+package middleware
+
+import (
+	"net/http"
+	"net/url"
+)
+
+// Redirect middleware, will redirect requests to hosts which match any of the
+// Matches to RedirectScheme://RedirectHost
+type Redirect struct {
+	Matches []Match
+
+	RedirectHost   string
+	RedirectScheme string
+}
+
+// Match specifies a match for a redirect.  Host and/or Scheme can be empty
+// signifiy match-all.
```

Comment:
It strikes me that `Host` can be blank. This api leaves no ability to match on blank hosts only. Or HTTP 1.0 calls without a Host header for that matter (which could probably be considered the same).
I don't see a nice way to resolve this though, so unless you feel strongly about it I'm prepared to say YAGNI.